phpBB/viewonline.php

```diff
@@ -32,491 +33,30 @@
-	extract($phpbb_dispatcher->trigger_event('core.viewonline_overwrite_location', compact($vars)));
-
-	$template_row = array(
-		'USERNAME' 			=> $row['username'],
-		'USERNAME_COLOUR'	=> $row['user_colour'],
-		'USERNAME_FULL'		=> $username_full,
-		'LASTUPDATE'		=> $user->format_date($row['session_time']),
-		'FORUM_LOCATION'	=> $location,
-		'USER_IP'			=> ($auth->acl_get('a_')) ? (($mode == 'lookup' && $session_id == $row['session_id']) ? gethostbyaddr($row['session_ip']) : $row['session_ip']) : '',
-		'USER_BROWSER'		=> ($auth->acl_get('a_user')) ? $row['session_browser'] : '',
-
-		'U_USER_PROFILE'	=> ($row['user_type'] != USER_IGNORE) ? get_username_string('profile', $row['user_id'], '') : '',
-		'U_USER_IP'			=> append_sid("{$phpbb_root_path}viewonline.$phpEx", 'mode=lookup' . (($mode != 'lookup' || $row['session_id'] != $session_id) ? '&amp;s=' . $row['session_id'] : '') . "&amp;sg=$show_guests&amp;start=$start&amp;sk=$sort_key&amp;sd=$sort_dir"),
-		'U_WHOIS'			=> append_sid("{$phpbb_root_path}viewonline.$phpEx", 'mode=whois&amp;s=' . $row['session_id']),
-		'U_FORUM_LOCATION'	=> $location_url,
-
-		'S_USER_HIDDEN'		=> $s_user_hidden,
-		'S_GUEST'			=> ($row['user_id'] == ANONYMOUS) ? true : false,
-		'S_USER_TYPE'		=> $row['user_type'],
+	$response = new RedirectResponse(
+		$controller_helper->route('phpbb_members_online_whois', array(
+			's'	=> $session_id,
+		), false),
+		301
 	);
-
-	/**
-	* Modify viewonline template data before it is displayed in the list
-	*
-	* @event core.viewonline_modify_user_row
-	* @var	array	on_page			File name and query string
-	* @var	array	row				Array with the users sql row
-	* @var	array	forum_data		Array with forum data
-	* @var	array	template_row	Array with template variables for the user row
-	* @since 3.1.0-RC4
-	*/
-	$vars = array('on_page', 'row', 'forum_data', 'template_row');
-	extract($phpbb_dispatcher->trigger_event('core.viewonline_modify_user_row', compact($vars)));
-
-	$template->assign_block_vars('user_row', $template_row);
+	$response->send();
 }
-$db->sql_freeresult($result);
-unset($prev_id, $prev_ip);
 
-$order_legend = ($config['legend_sort_groupname']) ? 'group_name' : 'group_legend';
-// Grab group details for legend display
-if ($auth->acl_gets('a_group', 'a_groupadd', 'a_groupdel'))
-{
-	$sql = 'SELECT group_id, group_name, group_colour, group_type, group_legend
-		FROM ' . GROUPS_TABLE . '
-		WHERE group_legend > 0
-		ORDER BY ' . $order_legend . ' ASC';
-}
-else
-{
-	$sql = 'SELECT g.group_id, g.group_name, g.group_colour, g.group_type, g.group_legend
-		FROM ' . GROUPS_TABLE . ' g
-		LEFT JOIN ' . USER_GROUP_TABLE . ' ug
-			ON (
-				g.group_id = ug.group_id
-				AND ug.user_id = ' . $user->data['user_id'] . '
-				AND ug.user_pending = 0
-			)
-		WHERE g.group_legend > 0
-			AND (g.group_type <> ' . GROUP_HIDDEN . ' OR ug.user_id = ' . $user->data['user_id'] . ')
-		ORDER BY g.' . $order_legend . ' ASC';
-}
-$result = $db->sql_query($sql);
 
```

Comment:
Unneeded whitespace. Why not do this?
```
if ($mode == 'whois')
{
	$response = new RedirectResponse(
		$controller_helper->route('phpbb_members_online_whois', array(
			's'	=> $session_id,
		), false),
		301
	);
}
else
{
	$response = new RedirectResponse(
		$controller_helper->route('phpbb_members_online', array(
			'mode'	=> $mode,
			's'		=> $session_id,
			'start'	=> $start,
			'sk'	=> $sort_key,
			'sd'	=> $sort_dir,
			'sg'	=> $show_guests,
		), false),
		301
	);
}

$response->send();
```

Reply:
Also, your redirect adds some "unwanted" infos that did not exist in the original URL:
`app.php/online?mode=&s=&start=0&sk=b&sd=d&sg=0`
When just entering viewonline.php that is.

Reply:
About the first thing, it is the same, but I it was how it was in viewonline.php, about the second thing should be fixed now